gobblin-runtime/src/main/java/gobblin/runtime/Task.java

```diff
@@ -392,4 +396,22 @@ private boolean inMultipleBranches(List<Boolean> branches) {
     }
     return inBranches > 1;
   }
+
+  /**
+   * Get the final state of each construct used by this task and add it to the {@link gobblin.runtime.TaskState}.
+   * @param extractor {@link gobblin.instrumented.extractor.InstrumentedExtractorBase} used by this task.
+   * @param converter {@link gobblin.converter.Converter} used by this task.
+   */
+  private void addConstructsFinalStateToTaskState(InstrumentedExtractorBase<?, ?> extractor,
+      Converter<?, ?, ?, ?> converter) {
+    this.taskState.addFinalConstructState(Constructs.EXTRACTOR.toString().toLowerCase(), extractor.getFinalState());
+    this.taskState.addFinalConstructState(Constructs.CONVERTER.toString().toLowerCase(), converter.getFinalState());
+    int forkIdx = 0;
+    for(Optional<Fork> fork : this.forks) {
+      if(fork.isPresent()) {
+        this.taskState.addFinalConstructState("fork." + forkIdx, fork.get().getFinalState());
```

Comment:
This is not a FORK_OPERATOR, it is a Fork, which is not part of gobblin-core. I will add the constant somewhere else though.